src/vs/workbench/api/common/extHost.api.impl.ts

```diff
@@ -526,6 +526,9 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 			},
 			registerInlineCompletionItemProvider(selector: vscode.DocumentSelector, provider: vscode.InlineCompletionItemProvider): vscode.Disposable {
 				checkProposedApiEnabled(extension, 'inlineCompletions');
+				if (provider.handleDidShowCompletionItem && !isProposedApiEnabled(extension, 'inlineCompletionsAdditions')) {
+					throw new Error(`When the method "handleDidShowCompletionItem" is implemented on a provider, the usage of the proposed api 'inlineCompletionsAdditions' must be declared!`);
+				}
```

Comment:
Prefer to use the standard error that is thrown from `checkProposedApiEnabled`. That includes instructions how to enable a proposal. Alternatively, only warn about the `handleDidShowCompletionItem`-property and then ignore it